spec/latest/json-ld-framing/index.html

```diff
@@ -552,34 +543,109 @@
+    <dt><code>[<a>IRI</a>+]</code></dt>
+    <dd>One or more strings in the form of an IRI, used for matching on <code>@type</code> and <code>@id</code>.</dd>
+    <dt><code>[<a>node object</a>]</code> (<code><dfn>node pattern</dfn></code>)</dt>
+    <dd>A non-empty <a>node object</a>, used to match specific values using recursive node matching.</dd>
+    <dt><code>[<a>value object</a>]</code> (<code><dfn>value pattern</dfn></code>)</dt>
+    <dd>A <a>value object</a>, used to match a specific value. Within a <a>value object</a>,
+      the values for <code>@value</code>, <code>@type</code>, and <code>@language</code>
+      may also be an array of one or more <a>string</a> values.</dd>
+  </dl>
+
+<p>The frame matching algorithm takes the <a>framing state</a> (<em>state</em>),
+  a list of subjects to match from the <a>map of flattened subjects</a> (<em>subjects</em>),
+  a <a>frame</a> to match against (<em>frame</em>), and the <em>requireAll</em> flag
+  and returns a list of matched subjects by filtering each <em>node</em> in <em>subjects</em> as follows:</p>
+
+<p>Frame matching follows a cascade, first attempting to match on a particular <code>@id</code>, then
+  a particular <code>@type</code> (or lack of <code>@type</code>), then by matching on any or all
+  of a set of properties, if neither <code>@id</code>, nor <code>@type</code> are in the frame.</p>
+
+<ol class="algorithm">
+  <li><em>Node</em> matches if it has an <code>@id</code> property including any <a>IRI</a> in the <code>@id</code> property in <em>frame</em>.</li>
+  <li><em>Node</em> matches if frame has no non-<code>keyword</code> properties.</li>
+  <li>If <em>requireAll</em> is <strong>true</strong>, <em>node</em> matches if all non-<a>keyword</a> properties (<em>property</em>) in <em>frame</em> match any of the following conditions.
+    Or, if <em>requireAll</em> is <strong>false</strong>, if any of the non-<a>keyword</a> properties (<em>property</em>) in <em>frame</em> match any of the following conditions.
+    For the <em>values</em> of each <em>property</em> from <em>frame</em> in <em>node</em>:
+    <ol class="algorithm">
+      <li>If <em>property</em> is <code>@type</code>:
+        <ol class="algorithm">
+          <li><em>Property</em> matches if the <code>@type</code> property in frame includes any <a>IRI</a> in <em>values</em>.</li>
+          <li>Otherwise, <em>property</em> matches if <em>values</em> is not empty and the <code>@type</code> property in <em>frame</em> is <code><a>wildcard</a></code>.</li>
+          <li>Otherwise, <em>property</em> matches if <em>values</em> is empty and the <code>@type</code> property in <em>frame</em> is <code><a>match none</a></code>.</li>
+          <li>Otherwise, <em>property</em> does not match.</li>
+        </ol>
+      </li>
+      <li>Otherwise, the value of <em>property</em> in <em>frame</em> MUST be empty, or an array
+        containing a valid <a>frame</a>.</li>
```

Comment:
Ok. Fine for now, but we may want to be more explicit with telling people to throw errors (and which "type").